release/Magarena/scripts/Herald_of_Secret_Streams.groovy

```diff
@@ -0,0 +1,16 @@
+[
+    new MagicStatic(
+        MagicLayer.Ability,
```

Comment:
This ability should be at the MagicLayer.Game layer as it is changing the game rules. In the modGame method, it should iterate over the permanents that satisfy the condition and add the Unblockable to them. As an example, see the groovy code for Ensnaring Bridge.

Reply:
Got it. Thanks for the confirmation!

Reply:
I read Ensnaring Bridge's groovy code but still don't quite understand what MagicTargetFilter.filter() methods do. Could you elaborate?

Reply:
Sure, the filter returns a list of permanents satisfying the filter's condition. So there are two steps here, create a filter, then call its filter method passing in the source. This returns a list which we iterate over using each.

Reply:
In this case, we already have a filter factory method you can use its `creature(MagicCounterType.PlusOne, Control.You)` this returns the filter you need.

Reply:
Hmm.. so what is source argument in modGame? Why does it use filter(source)?

Reply:
I see. Thanks for the explanation. Fixes for both cards pushed.